aw-server/src/android/mod.rs

```diff
@@ -93,33 +103,49 @@ pub mod android {
         string_to_jstring(&env, obj.to_string())
     }
 
+    lazy_static! {
+        static ref ASSET_PATH: RwLock<String> = RwLock::new(String::new());
+    }
+
     #[no_mangle]
     pub unsafe extern "C" fn Java_net_activitywatch_android_RustInterface_startServer(
         env: JNIEnv,
         _: JClass,
         java_asset_path: JString,
     ) {
-        use crate::config::AWConfig;
-        use std::path::PathBuf;
+        info!("Starting server...");
+
+        *ASSET_PATH.write().unwrap() = jstring_to_string(&env, java_asset_path);
+
+        // endpoints::build_rocket(server_state, config).launch().await;
+        start_server();
 
-        use crate::endpoints;
+        info!("Server exited");
+    }
 
+    #[rocket::main]
+    async fn start_server() {
         info!("Building server state...");
 
-        let asset_path = jstring_to_string(&env, java_asset_path);
-        info!("Using asset dir: {}", asset_path);
+        unsafe {
```

Comment:
I don't really understand why unsafe is needed here at all? Asked @johan-bjareholt and he was confused too.

```suggestion
        // FIXME: Why is unsafe needed here? Can we get rid of it?
        unsafe {
```

Maybe this inline comment would be good enough for now?

Reply:
@ErikBjare its because the openDatastore() is marked unsafe, hence it spirals down to other vars